pennylane_qiskit/vqe_runtime_runner.py

```diff
@@ -0,0 +1,488 @@
+        if len(tape.operations) == 0:
+            raise qml.QuantumFunctionError("Function contains no quantum operations.")
+
+        params = tape.get_parameters()
+        trainable_params = set()
+
+        for p in params:
+            if qml.math.requires_grad(p):
+                trainable_params.add(p)
+
+        num_params = len(trainable_params)
+
+        if len(x0) != num_params:
+            warnings.warn("Due to the tape expansion, the number of parameters has increased.")
+            x0 = 2 * np.pi * np.random.rand(num_params)
+
+        wires_circuit = tape.wires
+        wires_hamiltonian = hamiltonian.wires
+        all_wires = wires_circuit + wires_hamiltonian
+
+        # Set the number of qubits
+        num_qubits = len(all_wires)
+
+        consecutive_wires = qml.wires.Wires(range(num_qubits))
+        wires_map = OrderedDict(zip(all_wires, consecutive_wires))
+
+        # From here: Create the Qiskit ansatz circuit
+        params_vector = ParameterVector("p", num_params)
+
+        reg = QuantumRegister(num_qubits, "q")
+        circuit_ansatz = QuantumCircuit(reg, name="vqe")
+
+        circuits = []
+
+        j = 0
+        for operation in tape.operations:
+            wires = operation.wires.map(wires_map)
+            par = operation.parameters
+            operation = operation.name
+            mapped_operation = QiskitDevice._operation_map[operation]
+
+            qregs = [reg[i] for i in wires.labels]
+
+            if operation.split(".inv")[0] in ("QubitUnitary", "QubitStateVector"):
+                # Need to revert the order of the quantum registers used in
+                # Qiskit such that it matches the PennyLane ordering
+                qregs = list(reversed(qregs))
+
+            dag = circuit_to_dag(QuantumCircuit(reg, name=""))
+
+            if operation in ("QubitUnitary", "QubitStateVector"):
+                # Parameters are matrices
+                gate = mapped_operation(par[0])
+            else:
+                # Parameters for the operation
+                if par:
+                    # Trainable parameter
+                    if qml.math.requires_grad(par[0]):
+                        op_num_params = len(par)
+                        par = []
+                        for num in range(op_num_params):
+                            par.append(params_vector[j + num])
+                        j += op_num_params
+
+                gate = mapped_operation(*par)
+
+            if operation.endswith(".inv"):
+                gate = gate.inverse()
+
+            dag.apply_operation_back(gate, qargs=qregs)
+            circuit = dag_to_circuit(dag)
+            circuits.append(circuit)
+
+        for circuit in circuits:
+            circuit_ansatz &= circuit
+
+    else:
+        raise ValueError("Input ansatz is not a quantum function or a string.")
+
+    return x0, circuit_ansatz, num_qubits, all_wires
+
+
+def hamiltonian_to_list_string(hamiltonian, wires):
```

Comment:
@rmoyard working out the Codefactor issue would be best done now as later it will likely raise errors for future PRs. The Codefactor check is passing because another existing issue was resolved and Codefactor seems to care that the PR doesn't increase the number of issues in the codebase (resolving an issue and introducing a new balance out).